### ISSUE TITLE
m2e in Eclipse 2022-12 is updating Maven dependencies on every IDE start

### ISSUE THREAD

**Issue by ppalaga** [eclipse-m2e/m2e-core#1132]
... which leads to

* UI unresponsiveness
* an OOM Error after a few minutes.

![image](https://user-images.githubusercontent.com/1826249/206751622-1673aa2e-df89-4257-bfb6-46dbf51e8680.png)

```
java.util.concurrent.CompletionException: java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.OutOfMemoryError: Java heap space
	at org.eclipse.core.runtime.Path.toOSString(Path.java:1158)
	at org.eclipse.core.runtime.Path.toFile(Path.java:1117)
	at org.eclipse.lsp4e.LSPEclipseUtils.toUri(LSPEclipseUtils.java:935)
	at org.eclipse.lsp4e.LSPEclipseUtils.toUri(LSPEclipseUtils.java:945)
	at org.eclipse.lsp4e.LSPEclipseUtils.toWorkspaceFolder(LSPEclipseUtils.java:1002)
	at org.eclipse.lsp4e.LSPEclipseUtils$$Lambda$1136/0x0000000801a28ca0.apply(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at org.eclipse.lsp4e.LSPEclipseUtils.getWorkspaceFolders(LSPEclipseUtils.java:996)
	at org.eclipse.lsp4e.LanguageClientImpl.workspaceFolders(LanguageClientImpl.java:146)
	at org.eclipse.lsp4e.LanguageServerWrapper.getRelevantWorkspaceFolders(LanguageServerWrapper.java:225)
	at org.eclipse.lsp4e.LanguageServerWrapper.lambda$1(LanguageServerWrapper.java:293)
	at org.eclipse.lsp4e.LanguageServerWrapper$$Lambda$1096/0x0000000801a1cd80.apply(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	... 8 more
```

I have ~1400 Maven modules in my workspace and after years long experience with m2e, I do not expect this kind of operations to be fast or smooth. Therefore I am always trying to configure m2e not to do anything automatically. 

But here, I am running out of my skills. I'd expect the "Do not automatically update dependencies from remote repositories" to control exactly this, but having it on makes no difference. 

![image](https://user-images.githubusercontent.com/1826249/206752701-4211e9ed-3d7a-48d4-a651-e51e1efb5799.png)

Is there some other setting I can use to disable Updating Maven dependencies on every IDE start?

**Comment by HannesWell** [comment]
It looks like you have already disabled the automatic update of the Maven project configuration, which is enabled by default since Eclipse 2022-12.

Can you analyse (e.g. with VisualVM) what is consuming so much Memory?

The details of the first dialog you showed are the OOM Error?

@laeubi any idea? I see @ppalaga is working on Apache Camel. Is that also your 1400-projects workspace? Christoph, didn't you use Camel for some performance optimizations?

**Comment by laeubi** [comment]
I only see LSP4e in the stacktrace and nothing from m2e...

**Comment by ppalaga** [comment]
Thanks for your replies @HannesWell and @laeubi 

> Can you analyse (e.g. with VisualVM) what is consuming so much Memory?

Sorry, I cannot do that anymore, because the problem went away after I added `-Xmx4096m` to `eclipse.ini` and after I removed everything from the workspace and imported it again. Eclipse 2022-06 used to be content with `-Xmx2048m`. Maybe now after reimporting I can get back to `-Xmx2048m`.

What I have in this workspace is [Camel Quarkus](https://github.com/apache/camel-quarkus/).

I think we can close this one.

Thanks again for taking care!
